### PR TITLE
add `@guardian/libs` code

### PR DESCRIPTION
**Pull request by sndrs**
- adds the source code from `@guardian/libs` (https://github.com/guardian/libs)
- updates the `VisibilityState` type in the CWV tests to use TS's `DocumentVisibilityState`
- removes the old dev info from the readme
- updates `generateSvg.logger.teams.ts` script to match the CSNX node version, and accommodate changes in https://github.com/guardian/libs/pull/384 that were missed here

### CODE REVIEW

**Comment by sndrs** [comment]
@mxdvl are you ok with the changes in ede6453? (hard to spot them because the file is also added)

**Review by joecowton1 — APPROVED** [review]

File: libs/@guardian/libs/README.md [hunk 1]
```diff
@@ -104,24 +104,3 @@ If you are using this library with TypeScript, make sure you are using at least
 This package uses `ES2020`.
 
 If your target environment does not support that, make sure you transpile this package when bundling your application.
-
-## Development
```
[the remark this answers is not in the document]
Comment (sndrs) [reply libs/@guardian/libs/README.md:108]: no, this readme is the info gets displayed on npm for example, so it's still needed here

[the remark this answers is not in the document]
Comment (ob6160) [reply libs/@guardian/libs/README.md:108]: sorry — I was referring to the ones you removed

**Review by mxdvl — APPROVED** [review]
Thanks for updating the logger script!

File: libs/@guardian/libs/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sndrs) [reply libs/@guardian/libs/README.md:108]: no that's something we need to address, but probably beyond the scope of this PR?

**Comment by sndrs** [comment]
> Is it a problem at all that we have now lost git history?

we'll always have the history in the original repo – do you think we need to manage bringing in the history for the code that moves here? 

the conversations around this so far have tended towards concluding that history is useful to a point, but maybe not worth the noise it introduces in a migration like this (esp where the history starts again, but the earlier stages can be found in the old home)

File: libs/@guardian/libs/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ob6160) [reply libs/@guardian/libs/README.md:108]: okay, is it recorded somewhere that we need to address this? (on trello or similar)

**Comment by sndrs** [comment]
> Is it a problem at all that we have now lost git history?

maybe it would be useful to add a link to the original repo in the merge commit – that way if someone is diving into the history of these files, at least they'd eventually reach that commit that would redirect them to the old repo and it's history

**Comment by mxdvl** [comment]
> maybe it would be useful to add a link to the original repo in the merge commit – that way if someone is diving into the history of these files, at least they'd eventually reach that commit that would redirect them to the old repo and it's history

I think that would be very beneficial, along with a small explanation of the decision in the commit message and a link to the change log.

Contrary to many other repos, I would not expect people to rely on `blame` to delve into historical code, so a singular commit makes sense. Things that make this library unique are:
- 100% coverage
- extensive documentation
- strictest typescript checks
- versions and change log
- exports independent modules
- not driven by external business requirements

**Pull request merged**